### DISCUSSION TITLE
BasicCommand usage not adjustable

### DISCUSSION THREAD

**Discussion by Maloschnikow** [PaperMC/Paper#11493] [Ideas]
### Is your feature request related to a problem?

It's kind of related to a problem. I know you can set the command properties in plugin.yml, but since you can also set description within the code, I think it would be a great Idea to make the "Usage" property, which can beprinted via "/help command", also adjustable within the code.
For example I want to make a plugin with a simple /spawn command, that doesn't take any arguments. But in the "/help spawn" text it reads: "Usage: spawn [<args>]" and as far as I can see it there is no way except the plugin.yml to override this. (but maybe I'm just beeing dumb)

I also think there should be a "/" before the "spawn", because all the other commands have that.
![image](https://github.com/user-attachments/assets/d0d1e86b-476f-49f8-8c27-15902f25fa2e)

### Describe the solution you'd like.

The "Commands" interface has a function with several overloads to match the appropiate command registration, so add another overload to configure the Usage property.

### Describe alternatives you've considered.

It's possible to work around this with the plugin.yml, but I don't find this solution appealing.

**Comment by electronicboy** [comment]
Ther eis programatically no use case for modifying the Usage of a command outside of to other plugins accessing this stuff, at the point at which this occurs, this is far too late. if you want to setup stuff programmatically, define your commands programmatically and set the usage message there, that should apply to the help map, otherwise, you'd need to see if you can mess with the HelpMap, which wasn't really well designed for mutating stuff in there, and there is little interest from me in toying with the help map, it really should just go.

**Reply by Maloschnikow** [reply]
welp okay
